xarray/core/utils.py

```diff
@@ -62,8 +62,11 @@ def wrapper(*args, **kwargs):
 def _maybe_cast_to_cftimeindex(index: pd.Index) -> pd.Index:
     from ..coding.cftimeindex import CFTimeIndex
 
-    if index.dtype == 'O':
+    if len(index) > 0 and index.dtype == 'O':
```

Comment:
It might make more sense to instead check that `index` is not an instance of a `MultiIndex`.  I think the fact that `index.dtype` returns `object` in your example in #3000 is what trips things up originally.
```suggestion
    if not isinstance(index, pd.MultiIndex) and index.dtype == 'O':
```

Reply:
For the same reasons as above I wanted to avoid defaulting to CFTimeIndex([]) for other yet unknown cases